src/mod/endpoints/mod_sofia/sofia.c

```diff
@@ -1655,6 +1655,14 @@ static void our_sofia_event_callback(nua_event_t event,
 	}
 
 	if (sip && (status == 401 || status == 407)) {
+		if (channel) {
+			const char *sip_auth_username = switch_channel_get_variable(channel, "sip_auth_username");
+			const char *sip_auth_password = switch_channel_get_variable(channel, "sip_auth_password");
+			if (!zstr(sip_auth_username) && !zstr(sip_auth_password)) {
+				sofia_reg_handle_sip_r_challenge(status, phrase, nua, profile, nh, sofia_private, session, gateway, sip, de, tags);
+				goto done;
+			}
+		}
```

Comment:
For 407, I think it'll be better we hangup the call immediately with `MANDATORY_IE_MISSING` if no auth parameters are set rather having the sip session timeout itself

Reply:
Something like this?

Reply:
Yeah. Looks good now